src/compiler/types.ts

```diff
@@ -1273,7 +1275,15 @@ namespace ts {
 
     export interface TupleTypeNode extends TypeNode {
         kind: SyntaxKind.TupleType;
-        elementTypes: NodeArray<TypeNode>;
+        elements: NodeArray<TypeNode | NamedTupleMember>;

```

Comment:
In the end, I made `NamedTupleMember` a `TypeNode`, which is inline with `OptionalType` and `RestType`, so I don't _have_ to rename this field (or even change the type). If anyone feels strongly about it, I can change it back. However renaming it was very useful for figuring out where I needed to adjust/handle the new node; so "breaking" the API because of the new child kind might be worthwhile for other consumers, too. Depends on how strongly we want to maintain AST compatibility I suppose.

Reply:
I like the rename.